### PR TITLE
Add hovercard with cache request details

### PR DESCRIPTION
**Pull request by bduffany**
https://user-images.githubusercontent.com/2414826/166946187-27217920-25ed-4f33-8511-ca15ab0bf305.mp4

---

**Version bump**: Patch

### CODE REVIEW

**Comment by bduffany** [comment]
> the hovercard looks great, but i wonder if there's something we can do to make all of that blank space look more "hoverable"

Yeah, the hoverability isn't super obvious. fwiw, the whole row is hoverable -- I think since these rows take up so much of the screen real estate though, there's a pretty decent chance that people will hover over a row and at least discover it accidentally.

> cache requests are typically so short, i wonder if that "timeline" view is really the most useful. feel like i'd rather see the cache compression savings there or something

Yeah, I agree. The timeline seemed like a good idea until I started testing with larger invocations and limiting to mostly AC misses. Might just do away with it and put the raw data in the rows instead. Then maybe we could get rid of the need for a hovercard. wdyt?

**Comment by siggisim** [comment]
I'm cool with whatever you think makes sense here!

File: app/components/tooltip/tooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    el.style.top = `${top}px`;
+    el.style.opacity = "1";
+  }
+
+  private portalElement: HTMLElement | null = null;
+  private getPortalElement() {
+    if (this.portalElement) return this.portalElement;
+
+    // Note: all tooltips share a single portal.
+    let portal = document.querySelector(".tooltip-portal") as HTMLDivElement | null;
```
[the remark this answers is not in the document]
Comment (bduffany) [reply app/components/tooltip/tooltip.tsx]: Done

**Comment by bduffany** [comment]
> I'm cool with whatever you think makes sense here!

I'll send a follow-up to add more useful data to the rows, and hide the timeline behind a media query. I think on a large screen, the timeline can convey useful info when looking at CAS requests and sorting by duration.

**Pull request merged**